hera_mc/mc_session.py

```diff
@@ -1040,6 +1040,195 @@ def get_roach_temperature(self, starttime, stoptime=None, roach=None):
                                  stoptime=stoptime, filter_column='roach',
                                  filter_value=roach)
 
+    def add_node_sensor(self, time, node, top_sensor_temp, middle_sensor_temp,
+                        bottom_sensor_temp, humidity_sensor_temp, humidity):
+        """
+        Add new node sensor data to the M&C database.
+
+        Parameters:
+        ------------
+        time: astropy time object
+            astropy time object based on a timestamp reported by node
+        node: integer
+            node number (integer running from 1 to 30)
+        top_sensor_temp: float
+            temperature of top sensor reported by node in Celcius
+        middle_sensor_temp: float
+            temperature of middle sensor reported by node in Celcius
+        bottom_sensor_temp: float
+            temperature of bottom sensor reported by node in Celcius
+        humidity_sensor_temp: float
+            temperature of the humidity sensor reported by node in Celcius
+        humidity: float
+            humidity measurement reported by node
+        """
+        from .node import NodeSensor
+
+        self.add(NodeSensor.create(time, node, top_sensor_temp, middle_sensor_temp,
+                                   bottom_sensor_temp, humidity_sensor_temp, humidity))
+
+    def add_node_sensor_from_nodecontrol(self):
+        """Get and add node sensor information using a nodeControl object.
+        This function connects to the node and gets the latest data using the
+        `create_sensor` function.
+
+        If the current database is PostgreSQL, this function will use a
+        special insertion method that will ignore records that are redundant
+        with ones already in the database. This makes it convenient to sample
+        the node sensor data densely on qmaster.
+
+        """
+        from .node import create_sensor
+
+        node_sensor_list = create_sensor()
+
+        if self.bind.dialect.name == 'postgresql':
```

Comment:
Perhaps this logic could be split into a generic method, if it's going to appear in multiple places?

Reply:
yes it should!